### ISSUE TITLE
IDE0028 False positive for new expressions as a class property

### ISSUE THREAD

**Issue by rogerfar** [dotnet/roslyn#70952]
**Version Used**: VS 2022 17.8, .NET 8, C# latest

**Steps to Reproduce**:

```
public class Test
{
    private static readonly List<String> MyStringList = new()
    {
        "Item 1",
        "Item 2"
    };
}
```

**Diagnostic Id**: IDE0028 Collection initialization can be simplified

**Expected Behavior**:
This collection cannot be further simplified.

**Actual Behavior**:
It shows the error.

**Comment by CyrusNajmabadi** [comment]
This is by design.  That collection can be simplified further (to `["Item1", "Item2"]`).
